app/demo-loop/LDemoIO.root.cc

```diff
@@ -7,15 +7,64 @@
 //---------------------------------------------------------------------------//
 #include "LDemoIO.hh"
 
+#include <string>
+#include <vector>
 #include <TBranch.h>
 #include <TFile.h>
 #include <TTree.h>
 
+#include "celeritas/global/ActionRegistry.hh"
+
 namespace demo_loop
 {
 //---------------------------------------------------------------------------//
+/*!
+ * Store CoreParams data to the ROOT MC truth output file.
+ * Called by `to_root(...)`.
+ *
+ * \note
+ * Currently only storing the action labels so their IDs can be
+ * identified. If other parameters are needed for future
+ * debugging/analyses, this function can easily be expanded.
+ */
+void store_core_params(std::shared_ptr<celeritas::RootFileManager>& root_manager,
+                       celeritas::CoreParams core_params)
+{
+    auto action_reg = core_params.action_reg();
+    CELER_EXPECT(action_reg);
+
+    // Initialize CoreParams TTree
+    auto tree_params = root_manager->make_tree("core_params", "core_params");
+
+    // Store labels
+    std::vector<std::string> action_labels;
+    action_labels.resize(action_reg->num_actions());
+    for (auto const id : celeritas::range(action_reg->num_actions()))
+    {
+        action_labels[id] = action_reg->id_to_label(celeritas::ActionId{id});
+    }
+
+    // Set up action labels branch, fill the TTree and write it
+    /*
+     * The decision to store a vector instead of making a tree entry for
+     * each label is to simplify the reading of the information. Calling
+     * action_labels->at(action_id) after loading the first (and only) tree
+     * entry is much simpler than:
+     * tree->GetEntry(action_id);
+     * tree->GetLeaf("action_label")->GetValue();
+     */
+    tree_params->Branch("action_labels", &action_labels);
+    tree_params->Fill();
+    tree_params->Write();
+}
```

Comment:
Since this function is "private" (a local implementation function) it should go in an anonymous namespace.